org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/PropertyUtil.xtend

```diff
@@ -48,4 +50,42 @@ class PropertyUtil {
 		methodName.length > prefixLength && methodName.startsWith(prefix) && methodName.charAt(prefixLength).upperCase
 	}
 
+	/**
+	 * Returns the shorthand name for a function
+	 * e.g. URI for getURI; uri for setUri; uri for getuRI; enum for isEnum
+	 * If the given name is invalid, the result is <code>null</code>.
+	 * 
+	 * @param fullName the fullName of the function we're trying to shorthand (including 'get' or 'is')
+	 * 
+	 * @since 2.15
+	 */
+	/* @Nullable */
+	def static String tryGetShorthandName(String fullName) {
+		val name = if (fullName.startsWith("get") || fullName.startsWith("set")) {
+				fullName.substring(3)
+			} else if (fullName.startsWith("is")) {
+				fullName.substring(2)
+			} else {
+				return null
+			}
+
+		if (name.length() == 1) { // e.g. Point.getX()
+			return name.toLowerCase(Locale.ENGLISH);
+		} else if (name.length() > 1) {
+			if (Character.isUpperCase(name.charAt(1))) { // e.g. Resource.getURI
+				// if second char is uppercase, the name itself is the sugar variant
+				// URI is the property name for getURI
+				if (Character.isUpperCase(name.charAt(0))) {
+					return name;
+				}
+				// if the first character is not upper case, it's not a valid sugar variant
+				// e.g. uRI is no sugar access for getURI
+				return name;
```

Comment:
The code doesn't match the comment here. Should this return null instead?

Reply:
This comment appears to be copied from the previous utility, the return is correct but I'll fix the comment.